lib/dao.js

```diff
@@ -956,11 +956,22 @@ DataAccessObject._normalize = function (filter) {
 };
 
 function DateType(arg) {
-  var d = new Date(arg);
-  if (isNaN(d.getTime())) {
+  if (arg == null) {
+    return null;
+  } else if(typeof arg === 'object' && typeof arg.getTime === 'function') {
+    if (isNaN(arg.getTime())) {
+      throw new Error('Invalid date: ' + arg);
+    }
+    return arg;
+  } else if(typeof arg === 'string') {
+    var d = new Date(arg);
+    if (isNaN(d.getTime())) {
+      throw new Error('Invalid date: ' + arg);
+    }
+    return d;
+  } else {
     throw new Error('Invalid date: ' + arg);
   }
-  return d;
 }
```

Comment:
@raymondfeng it will not gonna work if you pass a "Date object"? In your code you just check if it is null, i have added 1. null check; 2. allow string and object for date.